src/AuthorizationHeaderBuilder.php

```diff
@@ -204,30 +204,18 @@ public function getAuthorizationHeader()
      *
      * The nonce is a v4 UUID.
      *
+     * @see https://stackoverflow.com/a/15875555
+     *
      * @return string
      *   The generated nonce.
      */
     public function generateNonce()
     {
-        return sprintf(
-            '%04x%04x-%04x-%04x-%04x-%04x%04x%04x',
-            // 32 bits for "time_low"
-            mt_rand(0, 0xffff),
-            mt_rand(0, 0xffff),
-            // 16 bits for "time_mid"
-            mt_rand(0, 0xffff),
-            // 16 bits for "time_hi_and_version",
-            // four most significant bits holds version number 4
-            mt_rand(0, 0x0fff) | 0x4000,
-            // 16 bits, 8 bits for "clk_seq_hi_res",
-            // 8 bits for "clk_seq_low",
-            // two most significant bits holds zero and one for variant DCE1.1
-            mt_rand(0, 0x3fff) | 0x8000,
-            // 48 bits for "node"
-            mt_rand(0, 0xffff),
-            mt_rand(0, 0xffff),
-            mt_rand(0, 0xffff)
-        );
+        $data = random_bytes(16);
```

Comment:
this is php7+, but this still supports php 5.5 at the moment doesn't it?

Reply:
Ah, missed that: we're currently depdending on [paragonie/random-compat](https://github.com/paragonie/random_compat) which has a random_bytes() shim so tests didn't fail. Will fix.